### PR TITLE
Events Backend Cleanup and Adding Start/End Range test

### PR DESCRIPTION
**Pull request by adam-james-v**
This is a small PR to add some tests to card and collection timeline endpoints.

As well, some ;; todo items are cleaned up.

### CODE REVIEW

File: test/metabase/api/collection_test.clj [hunk 1]
```diff
@@ -1380,6 +1380,60 @@
...
+
+(defn- event-names [timelines]
+  (->> timelines (mapcat :events) (map :name) set))
+
+(deftest timelines-test
+  (testing "GET /api/collection/root|id/timelines"
+    (mt/with-temp* [Collection [coll-a {:name "Collection A"}]
+                    Collection [coll-b {:name "Collection B"}]
+                    Collection [coll-c {:name "Collection C"}]]
+    (mt/with-temp* [Timeline [tl-a {:name          "Timeline A"
```
Comment (dpsutton) [inline test/metabase/api/collection_test.clj:1404]: You can just use a single `with-temp*` block, no need for the two nested ones.

Comment (adam-james-v) [reply test/metabase/api/collection_test.clj]: Good catch, thanks. Fixed here and in cards test too

File: src/metabase/api/timeline.clj [hunk 2]
```diff
@@ -15,13 +15,16 @@
   "Events Query Parameters Schema"
   (s/enum "events"))
 
+(def Icons
```
Comment (dpsutton) [inline src/metabase/api/timeline.clj:18]: I remember thinking about this location a bit. This seems like it specs the model not just the api. I think it should live in whichever model does not depend on the other model

File: test/metabase/api/timeline_test.clj [hunk 3]
```diff
@@ -127,7 +178,7 @@
                                                 :timeline_id (u/the-id timeline)
                                                 :archived    true}]]
           (testing "a timeline with no events returns an empty list"
-            (is (= #{} (event-names (include-events-request empty-tl false)))))
+            (is (= '() (:events (include-events-request empty-tl false)))))
```
Comment (dpsutton) [inline test/metabase/api/timeline_test.clj:181]: no need to change now but you don't need to quote the empty list in Clojure. And equality will check across collection type so you could use a more common `[]`. or just assert that it is `empty?`

**Review by dpsutton — APPROVED** [review]

**Comment by adam-james-v** [comment]
This was some cleanup for the Epic #20289

**Pull request merged**